File: PaCoSe.Models/DeviceConfig.cs
﻿using System.Collections.Generic;

namespace PaCoSe.Models
{
    public class DeviceConfig
    {
        public DeviceConfig()
        {
            this.DeviceLimits = new List<DeviceLimit>();
        }

        public string Name { get; set; }

        public bool IsScreenTimeEnabled { get; set; }

        public List<DeviceLimit> DeviceLimits { get; set; }
    }
}

File: PaCoSe.Web/ClientApp/src/app/common/models/device-config.ts
import { DeviceLimit } from "./deivce-limits";

export class DeviceConfig {
    public name: string;
    public isScreenTimeEnabled: boolean;
    public deviceLimits: DeviceLimit[];

    constructor(args) {
        this.name = args?.name;
        this.isScreenTimeEnabled = args?.isScreenTimeEnabled || false;
        this.deviceLimits = args?.deviceLimits?.length ? args.deviceLimits.map(l => new DeviceLimit(l)) : []
    }
}